1.18.11

matteobruni opened the pull request:
# Bug Fixes

- fix: opacity and size clamped, this should fix #1001
- fix: fixed preact definitions to fix #1003
- fix: fixed parent interactivity mode

# New Features

- feat: opacity animation options have now `startValue` and `destroy` properties like size animation options

matteobruni committed:
fix: opacity and size clamped, this should fix #1001

matteobruni committed:
feat: opacity animation options have now startValue and destroy like size animation options

ghost commented:
## DeepCode's analysis on [#782c07](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/681076e0e7d06d2dc1d8c3df263149260e98cd98/matteobruni/tsparticles/782c0717dd16eb33b9e421e19e8c71910cb7446a/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=1&) found:

- :information_source: **1** minor issue. :point_down:
- :heavy_check_mark: **1** issue was fixed.
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">'public' is implicit. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/782c0717dd16eb33b9e421e19e8c71910cb7446a/core/main/src/Core/Particles.ts#L19">Particles.ts:19</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/782c0717dd16eb33b9e421e19e8c71910cb7446a/core/main/src/Core/Particles.ts#L26">Particles.ts:26</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/782c0717dd16eb33b9e421e19e8c71910cb7446a/core/main/src/Core/Particles.ts#L27">Particles.ts:27</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/681076e0e7d06d2dc1d8c3df263149260e98cd98/matteobruni/tsparticles/782c0717dd16eb33b9e421e19e8c71910cb7446a/pr/_/%2Fcore%2Fmain%2Fsrc%2FCore%2FParticles.ts/javascript%2Ftslint%2Fmember-access/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/681076e0e7d06d2dc1d8c3df263149260e98cd98/matteobruni/tsparticles/782c0717dd16eb33b9e421e19e8c71910cb7446a/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=1&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=matteobruni)

matteobruni committed:
refactor: simplified destroy management on Updater.ts

matteobruni committed:
refactor: splitted some big methods

matteobruni committed:
fix: fixed preact definitions to fix #1003

matteobruni committed:
fix: mouse events works a lot better when interactivity element is not canvas

matteobruni committed:
fix: fixed parent interactivity mode

matteobruni committed:
feat: added new opacity options to editor

matteobruni committed:
Merge branch 'master' into r-1.18.11

matteobruni committed:
build: webpack configs for presets and shapes have now a common file

code-factor committed:
[CodeFactor] Apply fixes to commit d234d48

[ci skip] [skip ci]

matteobruni committed:
refactor(core): splitted particle links drawing from particle drawing

matteobruni committed:
docs: updated options for the animations of size and opacity

matteobruni committed:
chore(release): published new version

 - ng-particles@2.1.11
 - inferno-particles@1.1.11
 - jquery-particles@1.18.11
 - preact-particles@1.18.11
 - react-tsparticles@1.18.11
 - svelte-particles@1.1.11
 - particles.vue@2.1.11
 - particles.vue3@1.1.11
 - tsparticles-editor@1.18.11
 - tsparticles@1.18.11
 - ng-particles-demo@1.3.11
 - editor-demo@1.3.11
 - inferno-particles-demo@1.1.11
 - jquery-particles-demo@1.3.11
 - tsparticles-demo@1.3.11
 - preact-particles-demo@1.3.11
 - react-tsparticles-demo@1.3.11
 - svelte-demo@1.1.11
 - particles.vue-demo@1.3.11
 - particles.vue3-demo@1.1.11
 - tsparticles-preset-60fps@1.3.11
 - tsparticles-preset-background-mask@1.3.11
 - tsparticles-preset-basic@1.3.11
 - tsparticles-preset-big-circles@1.3.11
 - tsparticles-preset-bouncing@1.3.11
 - tsparticles-preset-fire@1.3.11
 - tsparticles-preset-font-awesome@1.3.11
 - tsparticles-preset-fountain@1.3.11
 - tsparticles-preset-snow@1.3.11
 - tsparticles-preset-stars@1.3.11
 - tsparticles-shape-bubble@1.3.11
 - tsparticles-shape-heart@1.3.11
 - tsparticles-shape-multiline-text@1.3.11
 - tsparticles-shape-spiral@1.3.11
 - cra-template-particles-typescript@1.1.11
 - cra-template-particles@1.1.11
